main/main.cpp

```diff
@@ -2086,10 +2090,14 @@ bool Main::iteration() {
 
 	bool exit = false;
 
-	Engine::get_singleton()->_in_physics = true;
-
 	for (int iters = 0; iters < advance.physics_steps; ++iters) {
 
+		if (InputDefault::get_singleton()->is_using_input_buffering() && agile_input_event_flushing) {
+			InputDefault::get_singleton()->flush_buffered_events();
+		}
+
+		Engine::get_singleton()->_in_physics = true;
```

Comment:
Are there any risks tied to toggling this on/off for each physics iteration like this?

Reply:
Just to decide one way or the other (since I don't see a strong reason for either), I decided this, so the input callbacks are still called with `Engine::is_in_physics_frame()` being `false`, to make it a bit more transparent to the game.

I can't see any potential risks.

Reply:
Yeah, it's hard to predict potential side effects, but it makes sense to keep input events reported as not being part of the physics tick, and we'll see if it causes issues in some edge cases by testing over time.